src/packages/core/components/input-tree/input-tree.element.ts

```diff
@@ -0,0 +1,130 @@
+import { css, html, customElement, property } from '@umbraco-cms/backoffice/external/lit';
+import { FormControlMixin } from '@umbraco-cms/backoffice/external/uui';
+import { UmbLitElement } from '@umbraco-cms/internal/lit-element';
+import { UmbInputDocumentElement } from '@umbraco-cms/backoffice/document';
+
+type NodeType = 'content' | 'member' | 'media';
+export type StartNode = {
+	type?: NodeType;
+	id?: string | null;
+};
+
+@customElement('umb-input-tree')
+export class UmbInputTreeElement extends FormControlMixin(UmbLitElement) {
+	protected getFormElement() {
+		return undefined;
+	}
+
+	private _type: StartNode['type'] = undefined;
+	@property()
+	public set type(newType: StartNode['type']) {
+		if (newType?.toLowerCase() !== this._type) {
+			this._type = newType?.toLowerCase() as StartNode['type'];
+		}
+	}
+	public get type(): StartNode['type'] {
+		return this._type;
+	}
+
+	@property({ type: String })
+	startNodeId?: string;
+
+	@property({ type: Number })
+	min = 0;
+
+	@property({ type: Number })
+	max = 0;
+
+	private _filter: Array<string> = [];
+	@property()
+	public set filter(value: string) {
+		this._filter = value.split(',');
+	}
+	public get filter(): string {
+		return this._filter.join(',');
+	}
+
+	@property({ type: Boolean })
+	showOpenButton?: boolean;
+
+	@property({ type: Boolean })
+	ignoreUserStartNodes?: boolean;
+
+	@property()
+	public set value(newValue: string) {
+		super.value = newValue;
+		if (newValue) {
+			this.selectedIds = newValue.split(',');
+		} else {
+			this.selectedIds = [];
+		}
+	}
+	public get value(): string {
+		return super.value as string;
+	}
+
+	selectedIds: Array<string> = [];
+
+	#onChange(event: CustomEvent) {
+		this.value = (event.target as UmbInputDocumentElement).selectedIds.join(',');
+		this.dispatchEvent(new CustomEvent('change', { bubbles: true, composed: true }));
+	}
+
+	constructor() {
+		super();
+	}
+
+	render() {
+		switch (this.type) {
+			case 'content':
+				return html`<umb-input-document
```

Comment:
You can simplify this code a bit because all of the properties are the same. If you programmatically create the correct input element, based on the type, then you can set the properties once on the element afterward. Something along the lines of:

```
let inputElement = null;
if (type === 'content') {
 inputElement = new UmbInputDocument();
} else if...

inputElement.selectedIds = this.selectedIds;

// insert your element into the dom
```